Add per-function filter exclusion option

Limess opened the pull request:
Allow users to set 
```yaml
logForwarding: false
``` 
on any given function. This excludes the function from the array of log group subscriptions generated.

This is useful in a few cases, e.g. a high traffic function which returns a static response.

It may be preferable to add `filterPattern` as an option on each function so a match-nothing pattern can be set, or have a new `logForwarding` object as an option on each function to make this extensible in future, although having a `false` boolean does not preclude this usage as a non-breaking change.

jconstance-amplify commented:
@Limess Thanks for the PR. I think I agree with you that I'd like to see this eventually expanded to a `logForwarding` object per function, but I think this is an okay start.

Can you update the CHANGELOG and README files with your changes?

Limess commented:
@jconstance-amplify I've moved the flag to `logForwarding.enabled` for each function to make it less ambiguous in future, and updated the changelog and README (I had originally done this but must have unstaged and removed it at some point!).
